src/main/java/com/github/rnewson/couchdb/lucene/util/Analyzers.java

```diff
@@ -186,13 +280,313 @@ protected TokenStreamComponents wrapComponents(String fieldName, TokenStreamComp
+		return fromSpec(json, Constants.ANALYZER);
+	}
+
+	/*
+	 * called from DatabaseIndexer when handling an http search request
+	 */
+	public static Analyzer fromSpec(String str) throws JSONException {
+		if (str == null) {
+			return getAnalyzer(Constants.DEFAULT_ANALYZER);
+		} 
+		
+		str = str.trim();
+		
+		if (str.startsWith("{")) {
+			try {
+				return getAnalyzer( new JSONObject(str) );
+			
+			} catch (Exception ex) {
+				return getAnalyzer( Constants.DEFAULT_ANALYZER );
+			}
+		}
+		
+		return getAnalyzer(str);
+	}
+
     public static Analyzer getAnalyzer(final String str) throws JSONException {
         final String[] parts = str.split(":", 2);
         final String name = parts[0].toUpperCase();
         final String args = parts.length == 2 ? parts[1] : null;
         return Analyzers.valueOf(name).newAnalyzer(args);
     }
 
+	public static Analyzer getAnalyzer(final JSONObject json) {
+		String className = json.optString(Constants.CLASS);
+		JSONArray params = json.optJSONArray(Constants.PARAMS);
+
+		Analyzer newAnalyzer = null;
+
+		if (className == null || className.isEmpty()) {
+			Iterator<?> it = json.keys();
+			if (it.hasNext()) {
+				String key = (String) it.next();
+				String args = json.optString(key);
+				
+				JSONObject obj = json.optJSONObject(key);
+				try {
+					if (obj != null) {
+						return Analyzers.valueOf(key.toUpperCase()).newAnalyzer(obj);
+					} else {
+						return Analyzers.valueOf(key.toUpperCase()).newAnalyzer(args);
+					}
+				} catch (Exception ex) { }
+			}
+
+			logger.error("Lucene index: analyzer class name is not defined");
+			return null;
+		}
+
+		// is the class accessible?
+		Class<?> clazz = null;
+		try {
+			clazz = Class.forName(className);
+		} catch (ClassNotFoundException e) {
+			logger.error(String.format("Lucene index: analyzer class %s not found. (%s)", className, e.getMessage()));
+			return null;
+		}
+
+		// Is the class an Analyzer?
+		if (!Analyzer.class.isAssignableFrom(clazz)) {
+			logger.error(String.format("Lucene index: analyzer class has to be a subclass of %s", Analyzer.class.getName()));
+			return null;
+		}
+
+		// Get list of parameters
+		List<KeyTypedValue> cParams;
+		try {
+			cParams = getAllConstructorParameters(params);
+		} catch (Exception pe) {
+			logger.error(String.format("Unable to get parameters for %s: %s", className, pe.getMessage()), pe);
+			cParams = new ArrayList<>();
+		}
+
+		// Iterate over all parameters, convert data to two arrays
+		// that can be used in the reflection code
+		final Class<?> cParamClasses[] = new Class<?>[cParams.size()];
+		final Object cParamValues[] = new Object[cParams.size()];
+		for (int i = 0; i < cParams.size(); i++) {
+			KeyTypedValue ktv = cParams.get(i);
+			cParamClasses[i] = ktv.getValueClass();
+			cParamValues[i] = ktv.getValue();
+		}
+
+		// Create new analyzer
+		newAnalyzer = createInstance(clazz, cParamClasses, cParamValues);
```

Comment:
rather than a method-scoped variable that is unused until the very end, an if statement here seems sufficient. tbh, given all we're doing is logging, which I think would be better done at the caller to this method anyway, I would just change this final section to 

`return createInstance(clazz, cParamClasses, cParamValues);`